.pre-commit-config.yaml

```diff
@@ -0,0 +1,15 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-added-large-files
+    -   id: check-byte-order-marker
+    -   id: check-case-conflict
+    -   id: check-executables-have-shebangs
+    -   id: check-shebang-scripts-are-executable
+    -   id: check-merge-conflict
+    -   id: mixed-line-ending
```

Comment:
it might be useful in future to set up a hook to run the Rust formatter, which does this as well as ensure that code is formatted uniformly according to recommended or configured style. same goes for trailing whitespace

Reply:
this is not immediately necessary though. will merge